pkg/noderesourcetopology/cache/indexer.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"strings"
+	"sync"
+
+	k8scache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+/*
+ * This is needed because of https://github.com/kubernetes/kubernetes/issues/110660
+ * Fixed in kubernetes 1.25.0, but enhancements are not backported.
+ * So we need to wait until the scheduler plugins repo consumes k8s >= 1.25.0, then
+ * We can just add a client-go/cache Indexer and get rid of this workaround:
```

Comment:
fair point. The indexer was born out of necessity but the concept kinda grew on me. We may want to at least discuss if we want to keep it around in the long run. 
That said, the challenge here is that the overallocating cache doesn't work with either our home-grown indexer or without using the `client-go` native solution. It would be great to always have the tree in a working state after each commit. I'll think about possible approaches here.

Reply:
I think I found a good approach which also trims down this pretty big commit. I'll push updates to this PR and only after fix #315